tests/ui/literal_string_with_formatting_arg.stderr

```diff
@@ -1,65 +1,77 @@
    |               ^^^
    |
-   = note: `-D clippy::literal-string-with-formatting-arg` implied by `-D warnings`
+   = note: `-D clippy::literal-string-with-formatting-args` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::literal_string_with_formatting_args)]`
 
 error: this looks like a formatting argument but it is not part of a formatting macro
-  --> tests/ui/literal_string_with_formatting_args.rs:8:16
+  --> tests/ui/literal_string_with_formatting_arg.rs:8:16
    |
 LL |     x.expect(" {y} bla");
    |                ^^^
 
 error: this looks like a formatting argument but it is not part of a formatting macro
-  --> tests/ui/literal_string_with_formatting_args.rs:9:15
+  --> tests/ui/literal_string_with_formatting_arg.rs:9:16
+   |
+LL |     x.expect(" {0}");
+   |                ^^^
+
+error: this looks like a formatting argument but it is not part of a formatting macro
+  --> tests/ui/literal_string_with_formatting_arg.rs:10:15
    |
 LL |     x.expect("{:?}");
    |               ^^^^
 
 error: this looks like a formatting argument but it is not part of a formatting macro
-  --> tests/ui/literal_string_with_formatting_args.rs:10:15
+  --> tests/ui/literal_string_with_formatting_arg.rs:11:15
    |
 LL |     x.expect("{y:?}");
    |               ^^^^^
 
 error: these look like formatting arguments but are not part of a formatting macro
-  --> tests/ui/literal_string_with_formatting_args.rs:11:16
+  --> tests/ui/literal_string_with_formatting_arg.rs:12:16
    |
 LL |     x.expect(" {y:?} {y:?} ");
    |                ^^^^^ ^^^^^
 
 error: this looks like a formatting argument but it is not part of a formatting macro
-  --> tests/ui/literal_string_with_formatting_args.rs:12:23
+  --> tests/ui/literal_string_with_formatting_arg.rs:13:23
    |
 LL |     x.expect(" {y:..} {y:?} ");
    |                       ^^^^^
 
 error: these look like formatting arguments but are not part of a formatting macro
-  --> tests/ui/literal_string_with_formatting_args.rs:13:16
+  --> tests/ui/literal_string_with_formatting_arg.rs:14:16
    |
 LL |     x.expect(r"{y:?}  {y:?} ");
    |                ^^^^^  ^^^^^
 
 error: this looks like a formatting argument but it is not part of a formatting macro
-  --> tests/ui/literal_string_with_formatting_args.rs:14:16
+  --> tests/ui/literal_string_with_formatting_arg.rs:15:16
    |
 LL |     x.expect(r"{y:?} y:?}");
    |                ^^^^^
 
 error: these look like formatting arguments but are not part of a formatting macro
-  --> tests/ui/literal_string_with_formatting_args.rs:15:19
+  --> tests/ui/literal_string_with_formatting_arg.rs:16:19
    |
 LL |     x.expect(r##" {y:?} {y:?} "##);
    |                   ^^^^^ ^^^^^
 
 error: this looks like a formatting argument but it is not part of a formatting macro
-  --> tests/ui/literal_string_with_formatting_args.rs:17:18
+  --> tests/ui/literal_string_with_formatting_arg.rs:18:18
    |
 LL |     x.expect("———{:?}");
```

Comment:
Why is this still being linted after restricting the lint to idents in the current scope?

Reply:
As long as there is a valid formatter with `:`, then it's highly likely that it's actually something the user intended. I don't know any language where `{:?}` or `{:#}` or `{:.2}` would make sense (doesn't mean they don't exist) so I think it's pretty safe to keep these cases. I can remove them if preferred but I think it's not cases that could trigger issues.

Reply:
That makes sense, I'm happy to keep it like this :D

Reply:
Yeay! \o/